src/main/java/com/yelp/nrtsearch/server/luceneserver/field/IndexableFieldDef.java

```diff
@@ -118,6 +120,20 @@ protected void validateRequest(Field requestField) {
               "field: %s cannot have both group and multivalued set to true. Cannot  group on multiValued fields",
               requestField.getName()));
     }
+    if (requestField.getKey()) {
+      if (requestField.getMultiValued()) {
```

Comment:
Could probably also verify that this is Keyable here as well.

Reply:
Good catch! I added the check now.